src/climate_data/generate/scenario_daily.py

```diff
@@ -173,113 +177,72 @@ def generate_scenario_daily_main(  # noqa: PLR0912, PLR0915, C901
-            v_anomaly = compute_anomaly(sref, target, anomaly_type)
-
-            key = f"{len(v_anomaly.latitude)}_{len(v_anomaly.longitude)}"
-
-            if key in source_anomalies:
-                old_count, old_anomaly = source_anomalies[key]
-
-                for coord in ["latitude", "longitude"]:
-                    old_c = old_anomaly[coord].to_numpy()
-                    new_c = v_anomaly[coord].to_numpy()
-                    tol = 1e-5
-
-                    if np.abs(old_c - new_c).max() < tol:
-                        v_anomaly = v_anomaly.assign({coord: old_c})
-                    else:
-                        msg = f"{coord} does not match despite having the same subdivision"
-                        raise ValueError(msg)
-                source_anomalies[key] = old_count + 1, old_anomaly + v_anomaly
-            else:
-                source_anomalies[key] = 1, v_anomaly
-        if source_anomalies:
-            anomalies[source] = source_anomalies
-
-    ensemble_anomaly = xr.Dataset()
-    for i, (source, source_anomalies) in enumerate(anomalies.items()):
-        sid = f"Source {i+1}/{len(source_paths)}: {source}"
-        print(f"Downscaling {i+1}/{len(anomalies)}: {source}")
-
-        source_ensemble_anomaly = xr.Dataset()
-        total_count = 0
-        for j, (res, (count, v_anomaly)) in enumerate(source_anomalies.items()):
-            res_id = f"{sid}, Resolution {j} / {len(source_anomalies)}: {res}"
-            print(f"Downscaling {res_id}")
-
-            if source_ensemble_anomaly.nbytes:
-                source_ensemble_anomaly += utils.interpolate_to_target_latlon(
-                    v_anomaly, method="linear"
-                )
-            else:
-                source_ensemble_anomaly = utils.interpolate_to_target_latlon(
-                    v_anomaly, method="linear"
-                )
-            total_count += count
-        source_ensemble_anomaly /= total_count
-
-        if ensemble_anomaly.nbytes:
-            ensemble_anomaly += source_ensemble_anomaly
-        else:
-            ensemble_anomaly = source_ensemble_anomaly
-
-    ensemble_anomaly /= len(anomalies)
-
-    print("Computing scenario data")
+    # randomly select source, variant, load reference and target,
+    # compute anomaly, resample anomaly and compute scenario data
+    source_key = random.choice(list(source_paths.keys()))
+    variant_paths = random.choice(source_paths[source_key])
+    s_variant = f"{variant_paths[0].stem.split('_')[-1]}"
+    vid = f"{source_key}, Variant : {s_variant}"
+    # load reference (monthly) and target (daily for a given year)
+    try:
+        print(f"{vid}: Loading reference")
+        sref = transform(*[load_variable(vp, "reference") for vp in variant_paths])
+        print(f"{vid}: Loading target")
+        target = transform(*[load_variable(vp, year) for vp in variant_paths])
+
+    except KeyError:
+        print(f"{vid}: Bad formatting, skipping...")
```

Comment:
I think since we're not outputting an ensemble, we just want to raise the error.  I.e. ditch the try-except block and just keep 188-191.  We do not want to start a workflow to produce 100 draws and have it complete without an error, but still produce fewer than 100 draws.